R/zchunk_L2112.ag_nonco2_IRR_MGMT.R

```diff
@@ -1,135 +1,193 @@
+    # used to  calculate share weights.
+    L2012.AgProduction_ag_irr_mgmt %>%
+      filter(year == max(BASE_YEARS)) %>%
+      mutate(AgProductionTechnology_nolvl = gsub("_hi|_lo", "", AgProductionTechnology)) %>%
+      group_by(region, AgSupplySector, AgSupplySubsector, AgProductionTechnology_nolvl, year) %>%
+      summarise(total = sum(calOutputValue)) %>%
+      ungroup(.) ->
+      L2112.AgProduction_ag_irr_nomgmt_aggergate
+
+    # Now subset the table of agricultural production for the most recent model base year
+    # and format the data frame so that so that the AgProductionTechnology_level column
+    # will match the column in the aggeregated data frame.
+    L2012.AgProduction_ag_irr_mgmt %>%
+      filter(year == max(BASE_YEARS)) %>%
+      select(region, AgSupplySector, AgSupplySubsector, AgProductionTechnology_lvl = AgProductionTechnology, year, calOutputValue) %>%
+      mutate(AgProductionTechnology_nolvl = gsub("_hi|_lo", "", AgProductionTechnology_lvl)) ->
+      L2112.AgProduction_ag
+
+    # Calculate the share weights or the fraction of emissions for each region, sector, subsector,
+    # technology and level for the using the aggrated total emissions determined above.
+    # Eventually these emission shares will be used as a factor to disaggregate the emissions input.
+    L2112.AgProduction_ag %>%
+      left_join_error_no_match(L2112.AgProduction_ag_irr_nomgmt_aggergate,
+                               by = c("region", "AgSupplySector", "year", "AgSupplySubsector", c("AgProductionTechnology_nolvl"))) %>%
+      mutate(share_tech = calOutputValue / total) ->
+      L2112.AgProduction_ag_share
+
+
+    # Combine non agricultural waste burning emissions and agricultural waste burning
+    # emissions into a single data frame. Add management level information to production
+    # technology column.
+    L2111.AWBEmissions %>%
+      bind_rows(L2111.AGREmissions) %>%
+      repeat_add_columns(tibble::tibble(level = c("hi", "lo"))) %>%
+      unite(AgProductionTechnology_lvl, AgProductionTechnology, level, sep = "_", remove = FALSE) ->
+      L2112.awb_agr_emissions
 
+    # Match the shares(fraction of emissions) to the data frames containing
+    # emissions quantities for the agricultural water burning and agricultural
+    # non waste burning emissions.
+    L2112.AgProduction_ag_share %>%
+      select(-year, -AgSupplySubsector) %>%
+      left_join(L2112.awb_agr_emissions, by = c("region", "AgSupplySector", "AgProductionTechnology_lvl")) ->
+      L2112.awb_agr_emissions
+
+
+    # Where shares allocated to lo/hi are NA but emissions are positive, split it 50/50 between
+    # the techs. For all others, set share to zero. Then scale the input emissions by the emission
+    # shares.
+    L2112.awb_agr_emissions %>%
+      mutate(share_tech = if_else(is.na(share_tech) & input.emissions > 1e-6, 0.5, share_tech)) %>%
+      mutate(share_tech = if_else(is.na(share_tech), 0, share_tech)) %>%
```

Comment:
Is it possible to use `replace_na` to replace something with multiple conditions? When I tried to use it to evaluate 2 statements I kept on getting `Evaluation error: no applicable method for 'replace_na' applied to an object of class "c('double', 'numeric')".`

If you really want me to used `replace_na` I would have to filter by the input.emission conditions, replace na, and then left join. Or I could do use `replace` for the first mutate statement and then `replace_na` in the second mutate statement. But this method will still use a `is.na`. Which method would prefer?